libsolutil/CommonIO.cpp

```diff
@@ -79,6 +79,16 @@ string solidity::util::readUntilEnd(istream& _stdin)
 	return ss.str();
 }
 
+string solidity::util::readBytes(istream& _input, size_t _length)
+{
+	string ret;
+	ret.resize(_length);
+	_input.read(ret.data(), static_cast<streamsize>(_length));
+	if (!_input)
```

Comment:
There are a few different failure conditions in iostreams and I always have to look them up to make sure I get this right. I'd add a comment explaining what's happening. Initially I thought it was wrong because you do not check `eofbit` but looks like `read()` sets `failbit` too on EOF so you do not have to.

I'd also check `fail()` explicitly so that it's not necessary to remember if it translates to `good()` or `!fail()` (they're not opposites of each other).

```suggestion
    // read() sets failbit in addition to eofbit if it reads fewer characters due to EOF
	if (_input.fail())
```